### PR TITLE
[spaceship] Fix login issues

### LINKED ISSUE #13281: [Regression] Build Fails with an Invalid 2 step response Error
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
When running a lane, the build is crashing after getting to the step where it tries to log into iTunes (App Store) Connect. As you can see in the stack trace, it gives an error about an "Invalid 2 step response" and the error message includes a JSON payload - I assume from Apple - with previously answered security questions. 

We didn't have this issue after updating to fastlane 1.102.0 on 8/29/18. I tried updating to fastlane 1.103.0, but am still running into the error.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
[15:41:27]: Running lane `ios development`. Next time you can do this by directly typing `fastlane ios development` 🚀.
[15:41:27]: Driving the lane 'ios development' 🚀
[15:41:27]: -----------------------
[15:41:27]: --- Step: cocoapods ---
[15:41:27]: -----------------------
[15:41:27]: $ pod install
[15:41:30]: ▸ [!] The version of CocoaPods used to generate the lockfile (1.4.0) is higher than the version of the current executable (1.3.1). Incompatibility issues may arise.
[15:41:46]: ▸ Analyzing dependencies
[15:41:46]: ▸ Downloading dependencies
[15:41:46]: ▸ Using AFNetworking (3.1.0)
[15:41:46]: ▸ Using CLTokenInputView (2.3.0)
[15:41:46]: ▸ Using Crashlytics (3.9.3)
[15:41:46]: ▸ Using EDColor (1.0.1)
[15:41:46]: ▸ Using EDSemver (0.4.0)
[15:41:46]: ▸ Using Expecta (1.0.6)
[15:41:46]: ▸ Using Fabric (1.7.2)
[15:41:46]: ▸ Using FirebaseAnalytics (3.9.0)
[15:41:46]: ▸ Using FirebaseCore (3.6.0)
[15:41:46]: ▸ Using FirebaseInstanceID (1.0.10)
[15:41:46]: ▸ Using Google (3.1.0)
[15:41:46]: ▸ Using GoogleAnalytics (3.17.0)
[15:41:46]: ▸ Using GoogleToolboxForMac (2.1.3)
[15:41:46]: ▸ Using LoopBack (1.3.0)
[15:41:46]: ▸ Using MBProgressHUD (1.1.0)
[15:41:46]: ▸ Using Mantle (2.1.0)
[15:41:46]: ▸ Using OCMock (3.4.1)
[15:41:46]: ▸ Using OHHTTPStubs (6.0.0)
[15:41:46]: ▸ Using ObjectiveSugar (1.1.0)
[15:41:46]: ▸ Using PromiseKit (4.5.2)
[15:41:46]: ▸ Using Realm (3.0.2)
[15:41:46]: ▸ Using Specta (1.0.7)
[15:41:46]: ▸ Using TestFairy (1.11.0)
[15:41:46]: ▸ Using UICKeyChainStore (2.1.1)
[15:41:46]: ▸ Using UIImage+PDF (1.1.3)
[15:41:46]: ▸ Using URBMediaFocusViewController (0.9.0)
[15:41:46]: ▸ Using google-cast-remote-display-sdk (2.10.4)
[15:41:46]: ▸ Using google-cast-sdk (2.10.4.1)
[15:41:46]: ▸ Using libextobjc (0.4.1)
[15:41:46]: ▸ Generating Pods project
[15:41:46]: ▸ Integrating client project
[15:41:47]: ▸ Sending stats
[15:41:47]: ▸ Pod installation complete! There are 25 dependencies from the Podfile and 29 total pods installed.
[15:41:47]: ▸ [!] The `XXXXXXX [Production]` target overrides the `DEBUG_INFORMATION_FORMAT` build setting defined in `Pods/Target Support Files/Pods-XXXXXXX/Pods-XXXXXXX.production.xcconfig'. This can lead to problems with the CocoaPods installation
[15:41:47]: ▸ [!] The `XXXXXXX [Development]` target overrides the `DEBUG_INFORMATION_FORMAT` build setting defined in `Pods/Target Support Files/Pods-XXXXXXX/Pods-XXXXXXX.development.xcconfig'. This can lead to problems with the CocoaPods installation
[15:41:47]: ▸ [!] The `XXXXXXXTests [Development]` target overrides the `DEBUG_INFORMATION_FORMAT` build setting defined in `Pods/Target Support Files/Pods-XXXXXXXTests/Pods-XXXXXXXTests.development.xcconfig'. This can lead to problems with the CocoaPods installation
[15:41:47]: ▸ [!] The `XXXXXXXTests [Production]` target overrides the `DEBUG_INFORMATION_FORMAT` build setting defined in `Pods/Target Support Files/Pods-XXXXXXXTests/Pods-XXXXXXXTests.production.xcconfig'. This can lead to problems with the CocoaPods installation
[15:41:47]: ▸ [!] The `XXXXXXXUITests [Development]` target overrides the `DEBUG_INFORMATION_FORMAT` build setting defined in `Pods/Target Support Files/Pods-XXXXXXXUITests/Pods-XXXXXXXUITests.development.xcconfig'. This can lead to problems with the CocoaPods installation
[15:41:47]: ▸ [!] The `XXXXXXXUITests [Production]` target overrides the `DEBUG_INFORMATION_FORMAT` build setting defined in `Pods/Target Support Files/Pods-XXXXXXXUITests/Pods-XXXXXXXUITests.production.xcconfig'. This can lead to problems with the CocoaPods installation
[15:41:47]: ▸ [!] Google has been deprecated
[15:41:47]: ▸ - Use the `$(inherited)` flag, or
[15:41:47]: ▸ - Remove the build settings from the target.
[15:41:47]: ▸ - Use the `$(inherited)` flag, or
[15:41:47]: ▸ - Remove the build settings from the target.
[15:41:47]: ▸ - Use the `$(inherited)` flag, or
[15:41:47]: ▸ - Remove the build settings from the target.
[15:41:47]: ▸ - Use the `$(inherited)` flag, or
[15:41:47]: ▸ - Remove the build settings from the target.
[15:41:47]: ▸ - Use the `$(inherited)` flag, or
[15:41:47]: ▸ - Remove the build settings from the target.
[15:41:47]: ▸ - Use the `$(inherited)` flag, or
[15:41:47]: ▸ - Remove the build settings from the target.
[15:41:47]: ------------------
[15:41:47]: --- Step: sigh ---
[15:41:47]: ------------------

+-------------------------------------+---------------------------+
|                    Summary for sigh 2.103.0                     |
+-------------------------------------+---------------------------+
| app_identifier                      | xxx.xxxxxxxxxxxx.ios      |
| username                            | xxxxxxx@xxxxxx.xxx        |
| provisioning_name                   | XXXX XXXXX XXXX XX XXXXX  |
|                                     | xxx.xxxxxxxxxxxx.ios      |
| adhoc                               | false                     |
| development                         | false                     |
| skip_install                        | false                     |
| force                               | false                     |
| team_id                             | XXXXXXXXXX                |
| ignore_profiles_with_different_name | false                     |
| skip_fetch_profiles                 | false                     |
| skip_certificate_verification       | false                     |
| platform                            | ios                       |
| readonly                            | false                     |
+-------------------------------------+---------------------------+

[15:41:47]: Starting login with user 'xxxxxxx@xxxxxx.xxx'
Service key is empty
[15:41:52]: -------------------
[15:41:52]: --- Step: slack ---
[15:41:52]: -------------------
[15:41:53]: Successfully sent Slack notification
+------------------+-----------------+
|            Lane Context            |
+------------------+-----------------+
| DEFAULT_PLATFORM | ios             |
| PLATFORM_NAME    | ios             |
| LANE_NAME        | ios development |
+------------------+-----------------+
[15:41:53]: Invalid 2 step response {"securityQuestions"=>{"questions"=>[{"id"=>135, "question"=>"What is the last name of your favorite elementary school teacher?", "number"=>1, "userDefined"=>false}, {"id"=>144, "question"=>"What is the name of the street where you grew up?", "number"=>3, "userDefined"=>false}]}, "crResetEnabled"=>false, "resetSecurityQuestionsSupportLink"=>"http://support.apple.com/kb/HT6170"}

+------+---------------------+-------------+
|             fastlane summary             |
+------+---------------------+-------------+
| Step | Action              | Time (in s) |
+------+---------------------+-------------+
| 1    | Verifying fastlane  | 0           |
|      | version             |             |
| 2    | default_platform    | 0           |
| 3    | cocoapods           | 19          |
| 💥   | sigh                | 5           |
| 5    | slack               | 0           |
+------+---------------------+-------------+

[15:41:53]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

/Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/two_step_client.rb:45:in `handle_two_step': [!] Invalid 2 step response {"securityQuestions"=>{"questions"=>[{"id"=>135, "question"=>"What is the last name of your favorite elementary school teacher?", "number"=>1, "userDefined"=>false}, {"id"=>144, "question"=>"What is the name of the street where you grew up?", "number"=>3, "userDefined"=>false}]}, "crResetEnabled"=>false, "resetSecurityQuestionsSupportLink"=>"http://support.apple.com/kb/HT6170"} (RuntimeError)
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/client.rb:463:in `send_shared_login_request'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/portal/portal_client.rb:28:in `send_login_request'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/client.rb:664:in `do_login'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/client.rb:363:in `login'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/client.rb:71:in `login'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/portal/spaceship.rb:25:in `login'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/spaceship/lib/spaceship/portal/spaceship.rb:103:in `login'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/sigh/lib/sigh/runner.rb:19:in `run'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/sigh/lib/sigh/manager.rb:8:in `start'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/actions/get_provisioning_profile.rb:19:in `run'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:40:in `block (2 levels) in parsing_binding'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /Users/ci/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.0/bin/fastlane:20:in `<top (required)>'
	from /Users/ci/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/ci/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'

  </pre>
</details>

### Environment

<details>
  <pre>
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.3                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.15.2 (Apple Git-101.1)        |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.3 (17D47)                    |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | true                                        |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.4.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
require 'json'

# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.56.0"

default_platform :ios

# Test Fairy URL
build_url = "https://XXXXXX.testfairy.com"

# Fixed build version minimum
build_version_minimum = "1.0.0"

platform :ios do
  before_all do
    cocoapods
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Compile a new Build for Development"
  desc "This will also make sure the profile is up to date"
  lane :development do

    sigh(
      app_identifier: "XXX.XXXXXXXXXXXX.XXX",
      username: "XXXXXXX@XXXXXX.XXX",
      output_path: "../../builds",
      provisioning_name: "XXXX XXXXX XXX XX XXXXX XXX.XXXXXXXXXXXX.ios"
    )

    # Add new-line to files. This can be handy for pods
    # sh "./add-newline.sh"

    gym(
      scheme: "Development",
      workspace: "XXXXXXX.xcworkspace",
      sdk: "iphoneos",
      clean: true,
      output_directory: "../../builds",
      output_name: "XXXXXXXXXXX-debug.ipa",
      export_method: "enterprise"
    )

    # desc "Upload to Test Fairy"
    testfairy(api_key: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
      ipa: "../../builds/XXXXXXXXXXX-debug.ipa",
      auto_update: "off",
      notify: "on",
      testers_groups: "XXXXXXX",
      comment: "XXXXXXXXXXX debug."
    )

    # desc "Build URL"
    build_url = lane_context[SharedValues::TESTFAIRY_BUILD_URL]
    
    # EX: 2017-11-08T16:27:29.582Z
    build_date = Time.new.strftime("%Y-%m-%dT%H:%M:%S%z")
    # Build number
    build_number = get_build_number(xcodeproj: "XXXXXXX.xcodeproj")
    # SEMVER version number
    build_version = get_version_number(xcodeproj: "XXXXXXX.xcodeproj")
    # version file dictionary
    versionHash = {
        "current" => build_version,
        "buildDate" => build_date,
        "minimum" => build_version_minimum
    }
    
    # Write the file to disk
    File.open("../../../builds/ios.version.json","w") do |f|
        f.write(versionHash.to_json)
    end
  end

  desc "Compile and Sanity Check a new Build for Production"
  desc "This will also make sure the profile is up to date"
  lane :sanitycheck do
    slack(
      message: "SANITYCHECK build started for XXX.XXXXXXXXXXXX.XXX to verify integrity."
    )

    sigh(
      app_identifier: "XXX.XXXXXXXXXXXX.ios",
      username: "XXXXXXX@XXXXXX.XXX",
      output_path: "../../builds",
      provisioning_name: "XXXX XXXXX XXX XX XXXXX XXX.XXXXXXXXXXXX.ios"
   )

    gym(
      scheme: "Production",
      workspace: "XXXXXXXXX.xcworkspace",
      sdk: "iphoneos",
      clean: true,
      output_directory: "../../builds",
      output_name: "XXXXXXXXXXX-sanitycheck.ipa",
      export_method: "enterprise",
      export_options: {
        manifest: {
          appURL: "https://XXXXXXXXXXXX.com/asset/downloads/XXXXXXXXXXX-sanitycheck.ipa",
          displayImageURL: "https://XXXXXXXXXXXX.com/asset/images/logo-XXXX.svg",
          fullSizeImageURL: "https://XXXXXXXXXXXX.com/asset/images/logo-XXXX.svg"
        }
      }
    )
    
    # desc "Rename the manifest file to the indended destination file"
    sh "mv ../../../builds/manifest.plist ../../../builds/XXXXXXXXXXX-sanitycheck.plist"

  end

  desc "Compile a new Build for Production"
  desc "This will also make sure the profile is up to date"
  lane :production do

    sigh(
      app_identifier: "XXX.XXXXXXXXXXXX.ios",
      username: "XXXXXXX@XXXXXX.XXX",
      output_path: "../../builds",
      provisioning_name: "XXXX XXXXX XXX XX XXXXX XXX.XXXXXXXXXXXX.ios"
    )

    # Add new-line to files. This can be handy for pods
    # sh "./add-newline.sh"

    gym(
      scheme: "Production",
      workspace: "XXXXXXXXX.xcworkspace",
      sdk: "iphoneos",
      clean: true,
      output_directory: "../../builds",
      output_name: "XXXXXXXXXXX-release.ipa",
      export_method: "enterprise",
      export_options: {
        manifest: {
          appURL: "https://XXXXXXXXXXXX.com/asset/downloads/XXXXXXXXXXX-release.ipa",
          displayImageURL: "https://XXXXXXXXXXXX.com/asset/images/logo-XXXX.svg",
          fullSizeImageURL: "https://XXXXXXXXXXXX.com/asset/images/logo-XXXX.svg"
          }
      }
    )

    # desc "Rename the manifest file to the indended destination file"
    sh "mv ../../../builds/manifest.plist ../../../builds/XXXXXXXXXXX-release.plist"

    # desc "Upload to Test Fairy"
    testfairy(api_key: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
      ipa: "../../builds/XXXXXXXXXXX-release.ipa",
      auto_update: "off",
      notify: "on",
      testers_groups: "XXXXXXXXXXXX, iOS-XXXXXXXXXXXX-Production",
      comment: "XXXXXXXXXXX release."
    )

    # desc "Build URL"
    build_url = lane_context[SharedValues::TESTFAIRY_BUILD_URL]
    
    # Version file information
    # For writing file version information
    # {
    #   "current": "1.0.0",
    #   "buildDate": "2017-11-08T16:27:29.582Z",
    #   "minimum": "1.0.0"
    # }
    
    # EX: 2017-11-08T16:27:29.582Z
    build_date = Time.new.strftime("%Y-%m-%dT%H:%M:%S%z")
    # Build number
    build_number = get_build_number(xcodeproj: "XXXXXXXXX.xcodeproj")
    # SEMVER version number
    build_version = get_version_number(xcodeproj: "XXXXXXXXX.xcodeproj")
    # version file dictionary
    versionHash = {
        "current" => build_version,
        "buildDate" => build_date,
        "minimum" => build_version_minimum
    }
    
    # Write the file to disk
    File.open("../../../builds/ios.version.json","w") do |f|
        f.write(versionHash.to_json)
    end
    
  end


  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    gym # Build your app - more options available
    pilot

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    # snapshot
    gym # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

     slack(
       message: "Successfully ran lane #{lane} with Test Fairy build URL #{build_url}",
       success: true
     )
  end

  error do |lane, exception|
     slack(
       message: exception.message,
       success: false
     )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "XXX.XXXXXXXXXXXX.ios" # The bundle identifier of your app
apple_id "XXXXXXX@XXXXXX.XXX" # Your Apple email address

team_id "XXXXXXXXXX"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.103.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| CFPropertyList            | 2.3.5        |
| activesupport             | 4.2.9        |
| addressable               | 2.5.1        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| bigdecimal                | 1.2.6        |
| bundler                   | 1.14.6       |
| claide                    | 1.0.2        |
| cocoapods                 | 1.3.1        |
| cocoapods-core            | 1.3.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| cocoapods-trunk           | 1.2.0        |
| cocoapods-try             | 1.1.0        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| commander-fastlane        | 4.4.6        |
| concurrent-ruby           | 1.0.5        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.0        |
| emoji_regex               | 0.1.1        |
| escape                    | 0.0.4        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.1.0        |
| fourflusher               | 2.0.1        |
| fuzzy_match               | 2.0.4        |
| gh_inspector              | 1.1.2        |
| google-api-client         | 0.21.2       |
| googleauth                | 0.5.1        |
| highline                  | 1.7.8        |
| http-cookie               | 1.0.3        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| i18n                      | 0.8.6        |
| io-console                | 0.4.3        |
| json                      | 1.8.1        |
| jwt                       | 1.5.6        |
| libxml-ruby               | 3.0.0        |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| memoist                   | 0.15.0       |
| mime-types                | 3.1          |
| mime-types-data           | 3.2016.0521  |
| mini_magick               | 4.5.1        |
| minitest                  | 5.4.3        |
| molinillo                 | 0.5.7        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| multipart-post            | 2.0.0        |
| nanaimo                   | 0.2.6        |
| nap                       | 1.1.0        |
| naturally                 | 2.1.0        |
| netrc                     | 0.7.8        |
| nokogiri                  | 1.5.6        |
| os                        | 0.9.6        |
| plist                     | 3.2.0        |
| power_assert              | 0.2.2        |
| psych                     | 2.0.8        |
| public_suffix             | 2.0.5        |
| rake                      | 10.4.2       |
| rdoc                      | 4.2.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| rouge                     | 2.0.7        |
| ruby-macho                | 1.1.0        |
| rubygems-update           | 2.5.2        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| signet                    | 0.7.3        |
| simctl                    | 1.6.4        |
| slack-notifier            | 2.3.2        |
| sqlite3                   | 1.3.13       |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.7.3        |
| test-unit                 | 3.0.8        |
| thread_safe               | 0.3.6        |
| tty-cursor                | 0.5.0        |
| tty-screen                | 0.6.3        |
| tty-spinner               | 0.8.0        |
| tzinfo                    | 1.2.3        |
| uber                      | 0.0.15       |
| unf                       | 0.1.4        |
| unf_ext                   | 0.0.7.4      |
| unicode-display_width     | 1.1.3        |
| word_wrap                 | 1.0.0        |
| xcodeproj                 | 1.6.0        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 0.0.4        |
</details>

*generated on:* **2018-09-04**
</details>

  </pre>
</details>

### LINKED ISSUE #13282: [Regression] [2.103.0] latest_testflight_build_number fails with privacy statement error
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Regression Description
Because we have `update_fastlane` in our Fastfile, our CI stopped working once 2.103.0 was released, with the following error log:
```
[19:42:45]: --- Step: latest_testflight_build_number ---
[19:42:45]: --------------------------------------------
[19:42:46]: Login to App Store Connect ([secure])
Service key is empty
+------------------+------------+
|         Lane Context          |
+------------------+------------+
| DEFAULT_PLATFORM | ios        |
| PLATFORM_NAME    | ios        |
| LANE_NAME        | ios deploy |
| VERSION_NUMBER   | 1.4.0      |
+------------------+------------+
[19:42:50]: Need to acknowledge to Apple's Apple ID and Privacy statement. Please manually log into https://appleid.apple.com (or https://appstoreconnect.apple.com) to acknowledge the statement.
[19:42:50]: fastlane finished with errors
```
Forcing the fastlane version back to 2.102.0 fixes the issue. Could this be related to 
#13272?

I am omitting further output for now in order to prevent posting sensitive information, but if this is not reproducible in other environments, let me know and I'll gladly share whatever is needed.

### LINKED ISSUE #13287: [Regression] Service key is empty
### Updated fastlane version 2.103.0

There was service key is empty I uploaded the app store.So I deleted the fastlane in project and reinstall.But :

15:52:23]: For more information, check out
[15:52:23]: 	https://github.com/fastlane/fastlane/tree/master/credentials_manager
[15:52:23]:
[15:52:23]: Please enter your Apple ID developer credentials
[15:52:23]: Apple ID Username:
****@****.com
[15:52:30]: Logging in...
Service key is empty
[15:52:39]: --------------------
[15:52:39]: fastlane init failed
[15:52:39]: --------------------
[15:52:39]: Invalid 2 step response {"securityQuestions"=>{"questions"=>[{"id"=>136, "question"=>"你******？", "number"=>2, "userDefined"=>false}, {"id"=>133, "question"=>"你****？", "number"=>1, "userDefined"=>false}]}, "crResetEnabled"=>false, "resetSecurityQuestionsSupportLink"=>"http://support.apple.com/kb/HT6170"}
[15:52:39]: Something failed while running `fastlane init`
[15:52:39]: Tried using Apple ID with email '****'
[15:52:39]: You can either retry, or fallback to manual setup which will create a basic Fastfile

### PR DESCRIPTION
**Pull request by cjweigle**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Logging in stopped working in 2.103.0, the url used when fetching the authServiceKey shouldn't have been changed in #13272. 

In addition: an environment variable ITC_ITUNES_ENDPOINT can be set to 1 to use the itunesconnect.apple.com. [#13282]

Also should fix #13281, ~~possibly #13284~~

### Description
I've reverted the url in itc_service_key, and added the ITC_ITUNES_ENDPOINT. I've built and uploaded a build to testflight. 

--- 
closes #13281
closes #13282 
closes #13287

### CODE REVIEW

File: spaceship/lib/spaceship/test_flight/client.rb [hunk 1]
```diff
@@ -7,15 +7,15 @@ class Client < Spaceship::Client
       # Spaceship HTTP client for the testflight API.
       #
       # This client is solely responsible for the making HTTP requests and
-      # parsing their responses. Parameters should be either named parameters, or
```
Comment (janpio) [inline spaceship/lib/spaceship/test_flight/client.rb]: huh?

File: spaceship/lib/spaceship/client.rb [hunk 2]
```diff
@@ -89,6 +89,10 @@ def teams
       end
     end
 
+    def self.base_url
+      ENV["ITC_ITUNES_ENDPOINT"] ? "itunesconnect.apple.com" : "appstoreconnect.apple.com"
```
Comment (janpio) [inline spaceship/lib/spaceship/client.rb]: Why the ENV variable? 
What problem does this solve that the endpoint can be switched?

**Comment by joshdholtz** [comment]
@chronweigle I'm going to push a commit here to remove that environment variable in `base_url` and also update some comments! Hope that's okay 😊

File: spaceship/lib/spaceship/client.rb (hunk 2 above)
Comment (janpio) [reply spaceship/lib/spaceship/client.rb]: #13282 was also caused by "Service key is empty", which was triggered by the wrong URL being used on the olympus.itunes.apple.com request. 1.103.1 should fix that.

**Comment by janpio** [comment]
Thanks so much for this PR @chronweigle, as you know @joshdholtz created another one that only contained the "olympus" fix and merged and released that one yesterday as 1.103.1. It was great waking up to the regression **and** a PR that fixes it already 💯 🥇 

(Can you maybe reply to my comment re the ENV var? Maybe this is still something we should consider doing - I just don't get the intentions behind it (yet)).

**Pull request closed without merging**